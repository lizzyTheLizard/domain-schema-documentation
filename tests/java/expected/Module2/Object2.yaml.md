# Object 2
![Static Badge](https://img.shields.io/badge/Validator%20Errors-3-red)

An Object in Module 2
```mermaid
classDiagram
namespace Module 2 {
  class _Module2_Object2_yaml["Object 2"]
  class Type["Type"]
  class AdditionalProperties["AdditionalProperties"]
}
namespace Module {
  class _Module_Object_yaml["Object"]
}
_Module_Object_yaml ..> _Module2_Object2_yaml :obj2
_Module2_Object2_yaml ..> Type :type
_Module2_Object2_yaml o-- AdditionalProperties 
click _Module2_Object2_yaml href "./Object2.yaml.html" "Object 2"
click Type href "./Object2.yaml.html" "Object 2"
click AdditionalProperties href "./Object2.yaml.html" "Object 2"
click _Module_Object_yaml href "../Module/Object.yaml.html" "Object"
```

## Examples
```json
{
  "id": "1",
  "type": "type1",
  "additional": {
    "key": "key",
    "value": "value"
  }
}
```



## Properties
| Name | Type | Description |
|------|------|-------------|
| id | String |  |
| type | [Type](#Type) |  |
| (other) | [AdditionalProperties](#AdditionalProperties) | Additional Properties |

## Subschemas
### Type (Enum)



| Name | Description |
|------|-------------|
| type1 | Description of type1 |
| type2 | Description of type2 |

### AdditionalProperties (Object)




| Name | Type | Description |
|------|------|-------------|
| key | String |  |
| value | String |  |

## Verification Errors
| Type | Description |
|------|-------------|
| MISSING_IN_IMPLEMENTATION | &#x27;com.example.module2.model.Object2&#x27; should exist but is missing in the implementation |
| MISSING_IN_IMPLEMENTATION | &#x27;com.example.module2.model.Object2Type&#x27; should exist but is missing in the implementation |
| MISSING_IN_IMPLEMENTATION | &#x27;com.example.module2.model.Object2AdditionalProperties&#x27; should exist but is missing in the implementation |

## Links
1. [Java-File](./java/Object2.java)
1. [Java-File (Type)](./java/Object2Type.java)
1. [Java-File (AdditionalProperties)](./java/Object2AdditionalProperties.java)
