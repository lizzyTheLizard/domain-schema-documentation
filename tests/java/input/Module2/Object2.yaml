$id: /Module2/Object2.yaml
title: Object 2
description: An Object in Module 2
type: object
x-schema-type: Other
examples:
  - id: "1"
    type: "type1"
    additional: 
      key: "key"
      value: "value"
properties:
  id:
    type: string
  type:
    type: string
    enum:
      - type1
      - type2
    x-enum-description:
      type1: Description of type1
      type2: Description of type2
additionalProperties: 
  type: object
  properties:
    key:
      type: string
    value:
      type: string

