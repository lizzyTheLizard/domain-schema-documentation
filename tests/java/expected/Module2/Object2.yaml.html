<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="Mark Otto, Jacob Thornton, and Bootstrap contributors">
    <meta name="generator" content="Hugo 0.84.0">
    <title>Test Application</title>
    <link rel="canonical" href="https://getbootstrap.com/docs/5.0/examples/navbar-static/">
    <link rel=stylesheet href=https://cdn.jsdelivr.net/npm/pretty-print-json@3.0/dist/css/pretty-print-json.css>
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css" rel="stylesheet">
    <meta name="theme-color" content="#7952b3">
    <style>
    .alert ul {
      margin-bottom: 0;
    }
    .mermaid{
      height: 300px;
      border: 1px solid #000;
    }
    .mermaid pre {
      display: none;
    }
    section {
      margin-bottom: 50px;
    }
    .title-badge {
      font-size: 0.3em;
      vertical-align: top;
    }
    .json {
      border: 1px solid black;
      background-color: lightgray;
      padding: 10px;
     }
    </style>
  </head>
  <body>
    <main class="container">
      <nav aria-label="breadcrumb">
        <ol class="breadcrumb">
          <li class="breadcrumb-item"><a href="../../index.html">Application</a></li>
          <li class="breadcrumb-item"><a href="./index.html">Module 2</a></li>
          <li class="breadcrumb-item active" aria-current="page">Object 2</li>
        </ol>
      </nav>
      
      <h1>Object 2</h1>
      <img alt="Static Badge" src="https://img.shields.io/badge/Validator%20Errors-3-red">
      
      
      <section>
        <p>An Object in Module 2</p>
        <div class="mermaid">
          <pre class="mermaidDefinition">
            classDiagram
      namespace Module 2 {
        class _Module2_Object2_yaml["Object 2"]
        class Type["Type"]
        class AdditionalProperties["AdditionalProperties"]
      }
      namespace Module {
        class _Module_Object_yaml["Object"]
      }
      _Module_Object_yaml ..> _Module2_Object2_yaml :obj2
      _Module2_Object2_yaml ..> Type :type
      _Module2_Object2_yaml o-- AdditionalProperties 
      click _Module2_Object2_yaml href "./Object2.yaml.html" "Object 2"
      click Type href "./Object2.yaml.html" "Object 2"
      click AdditionalProperties href "./Object2.yaml.html" "Object 2"
      click _Module_Object_yaml href "../Module/Object.yaml.html" "Object"
          </pre>
        </div>
      </section>
      
      
      
      <section>
        <h2>Properties</h2>
        <table class="table">
          <thead>
            <tr>
              <th>Name</th>
              <th>Type</th>
              <th>Description</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>id</td>
              <td>String</td>
              <td></td>
            </tr>
            <tr>
              <td>type</td>
              <td><a href="#Type">Type</a></td>
              <td></td>
            </tr>
            <tr>
              <td>(other)</td>
              <td><a href="#AdditionalProperties">AdditionalProperties</a></td>
              <td>Additional Properties</td>
            </tr>
          </tbody>
        </table>
      </section>
      
      <section>
        <h2>Examples</h2>
        <pre class="json">
        {"id":"1","type":"type1","additional":{"key":"key","value":"value"}}
        </pre>
      </section>
      
      <section>
        <h2>Verification Errors</h2>
        <table class="table">
          <thead>
            <tr>
              <th>Type</th>
              <th>Description</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>MISSING_IN_IMPLEMENTATION</td>
              <td>&#x27;com.example.module2.model.Object2&#x27; should exist but is missing in the implementation</td>
            </tr>
            <tr>
              <td>MISSING_IN_IMPLEMENTATION</td>
              <td>&#x27;com.example.module2.model.Object2Type&#x27; should exist but is missing in the implementation</td>
            </tr>
            <tr>
              <td>MISSING_IN_IMPLEMENTATION</td>
              <td>&#x27;com.example.module2.model.Object2AdditionalProperties&#x27; should exist but is missing in the implementation</td>
            </tr>
          </tbody>
        </table>
      </section>
      
      <section>
        <h2>Links</h2>
        <ul>
          <li><a href="./java/Object2.java">Java-File</a></li>
          <li><a href="./java/Object2Type.java">Java-File (Type)</a></li>
          <li><a href="./java/Object2AdditionalProperties.java">Java-File (AdditionalProperties)</a></li>
        </ul>
      </section>
      
      
      <h2>Subschemas</h2>
      <a id="Type"></a><h3>Type   (Enum)  </h3>
      <section>
        <p></p>
        <table class="table">
          <thead>
            <tr>
              <th>Name</th>
              <th>Description</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>type1</td>
              <td>Description of type1</td>
            </tr>
            <tr>
              <td>type2</td>
              <td>Description of type2</td>
            </tr>
          </tbody>
        </table>
      
      </section>
      <a id="AdditionalProperties"></a><h3>AdditionalProperties    (Object)</h3>
      <section>
        <p></p>
      
        <table class="table">
          <thead>
            <tr>
              <th>Name</th>
              <th>Type</th>
              <th>Description</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>key</td>
              <td>String</td>
              <td></td>
            </tr>
            <tr>
              <td>value</td>
              <td>String</td>
              <td></td>
            </tr>
          </tbody>
        </table>
      </section>
      
    </main>
    <script src="https://cdn.jsdelivr.net/npm/svg-pan-zoom@3.5.0/dist/svg-pan-zoom.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/js/bootstrap.bundle.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/pretty-print-json@3.0/dist/pretty-print-json.min.js"></script>
    <script type="module">
      import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@10/dist/mermaid.esm.min.mjs';

      function prettifyJson(){
        const elements = document.querySelectorAll('.json');
        for(const element of elements) {
          element.innerHTML = prettyPrintJson.toHtml(JSON.parse(element.innerText));
        }
      }

      function drawDiagrams() {
        const elements = document.querySelectorAll('.mermaid');
        elements.forEach((element, i) =>drawDiagram(element, i));
      }

      async function drawDiagram(element, i) {
        const graphDefinition = element.querySelector(".mermaidDefinition")
        const {svg} = await mermaid.render('mySvg_' + i, graphDefinition.innerText);
        const preview = createMermaidPreview(element, svg);
        const modal = createMermaidModal();
        //Do this with a slight delay to take link in mermaid diagram take precedence
        element.addEventListener('click', () => setTimeout(() => showMermaidModal(preview,modal), 30));
      }

      function createMermaidPreview(element, svg) {
        const preview = document.createElement('div');
        preview.style.height = '100%';
        preview.style.width = '100%';
        preview.innerHTML = svg.replace(/[ ]*max-width:[ 0-9\.]*px;/i , '');
        preview.querySelector("svg").style.height = '100%';
        preview.querySelector("svg").style.width = '100%';
        element.appendChild(preview);
        return preview;
      }

      function createMermaidModal() {
        const modal = document.createElement('div');
        modal.innerHTML =
            '<div class="modal" tabindex="-1">\n' +
            '  <div class="modal-dialog modal-fullscreen">\n' +
            '    <div class="modal-content">\n' +
            '      <div class="modal-header">\n' +
            '        <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>\n' +
            '      </div>\n' +
            '      <div class="modal-body"></div>\n' +
            '    </div>\n' +
            '  </div>\n' +
            '</div>';
        document.body.appendChild(modal);
        return modal;
      }

      function showMermaidModal(preview, modal) {
        const modalBody = modal.querySelector('.modal-body');
        modalBody.innerHTML = preview.innerHTML
        const modalDiv = modal.querySelector('.modal');
        const svg = modalBody.querySelector('svg');
        new bootstrap.Modal(modalDiv).show();
        svgPanZoom(svg, {
          zoomEnabled: true,
          controlIconsEnabled: true,
          fit: true,
          center: true
        })
      }

      mermaid.initialize({ startOnLoad: false,securityLevel: 'loose'});
      drawDiagrams();
      prettifyJson();
    </script>
  </body>
</html>
